### DISCUSSION TITLE
Debian ppc64 (970MP) build

### DISCUSSION THREAD

**Discussion by unsound** [rmottola/Arctic-Fox#173] [Show and tell]
Hi, I thought I'd post a build for Debian ppc64 (sid) (64k page size) that I did on my PowerMac G5 Dual 2.3 the other day. Maybe it can be useful to someone else:
https://github.com/unsound/Arctic-Fox/releases/tag/v43.0-build.debian-ppc64

I can't browse GitHub with it though, it goes all black and white and weird. Is this a known issue? Other less complicated sites work fine.

**Comment by unsound** [comment]
Updated to 43.1: https://github.com/unsound/Arctic-Fox/releases/tag/v43.1-build.debian-ppc64

**Comment by rmottola** [comment]
Github is a known cracker, it will not work on Intel either, it needs a very very recent browser.
With current upcoming v45  most pages are at least readable, so it gets closer to what e.g. SeaMonkey does,  but still not fully usable.

Unfortunately, it appears it will not work on Linux/PPC - do you care some test builds of the git version @unsound ?
It works on NetBSD though!

See https://github.com/rmottola/Arctic-Fox/issues/210

**Comment by rmottola** [comment]
If you keep up building, I fear you will encounter issues with 45.2. 45.1 should build and work, provided you use a system ICU. Share your experience.

**Comment by unsound** [comment]
I'll see if I can find time this week to attempt a build.

**Comment by unsound** [comment]
Ok, the results so far: 45.1 is able to build but still fails in the same way as 45.0 with the system ICU (`72.1-5+b1`). 45.2 doesn't build and fails at:
```
141:57.38 /ZStore/devel/Arctic-Fox/js/src/asmjs/WasmBaselineCompile.cpp:1032:36: error: ‘scratch’ was not declared in this scope
141:57.38  1032 |                 loadI64(Register64(scratch), v);
141:57.39       |                                    ^~~~~~~
```

**Comment by rmottola** [comment]
I don't fully get - System ICU should work (provided it is not too new), bundled ICU is little-endian only.
I imported extra 1286207 and 1286767! 
You might try current dev .... error chanced a little bit, but still doesn't work. Not it is consistent between PPC 32bit and 64bit though (if you can confirm).

https://github.com/rmottola/Arctic-Fox/issues/233

**Comment by unsound** [comment]
I'm sorry, I messed up when reviewing the ```.mozconfig```... I somehow thought that the system ICU was enabled but missed the ```#``` preceding it. :) So I rebuilt it after looking more carefully and I'm happy to report that the 45.1 build does work now! I uploaded the results here:
https://github.com/unsound/Arctic-Fox/releases/tag/v45.1-build.debian-ppc64

**Comment by rmottola** [comment]
Very good. I suppose 45.2 doesn't build for you. However I was able to fix jit-none builds. I was able to build PPC32 yesterday, but the browser starts shows a window and crashes. If you want to test dev branch, see how it fares for you.

**Comment by unsound** [comment]
Will do, I will build 45.1 for 32-bit powerpc first and then attempt the latest dev state.

Edit: 45.1 for powerpc also worked fine, no immediate issues. Uploaded:
https://github.com/unsound/Arctic-Fox/releases/tag/v45.1-build.debian-powerpc

**Comment by rmottola** [comment]
You should be able to compile 46.1 with the same setup too (actually you need one single patch as backport, which is in master). How does it work for you?

**Comment by unsound** [comment]
I compiled 46.2 successfully for ppc64/debian sid with the same parameters as the last build but it segfaults with the messages:
```
[664183] ###!!! ABORT: u_init() failed: file /ZStore/devel/Arctic-Fox/xpcom/build/XPCOMInit.cpp, line 710
[664183] ###!!! ABORT: u_init() failed: file /ZStore/devel/Arctic-Fox/xpcom/build/XPCOMInit.cpp, line 710
```

The contents around line 710:
```
  // Initialize the JS engine.                                                                                                                                 
  const char* jsInitFailureReason = JS_InitWithFailureDiagnostic();
  if (jsInitFailureReason) {
    NS_RUNTIMEABORT(jsInitFailureReason);
  }
```

I'm not sure what it's related to, I can debug more towards the weekend.

**Reply by rmottola** [reply]
@unsound how do you handle the ICU issue? You need to use an "old enough" good system ICU or susbistitute the dat file. https://github.com/rmottola/Arctic-Fox/issues/264  https://github.com/rmottola/Arctic-Fox/issues/203

**Comment by rmottola** [comment]
@unsound if you use up-to-date debian, there is the fontconfig #258  issue to consider, as well as ICU (trying to work around using external dat file).
I have refrained to update my PowerBook so I can continue building and got a starting AF with the upcoming 47
